### DISCUSSION TITLE
`Pull Command` directly displays the specific image name + label name

### DISCUSSION THREAD

**Discussion by ghost** [goharbor/harbor#19662] [Ideas]
Is it possible to directly add a function that allows users to copy the pulled command with a click like in dockerhub, instead of manually modifying it into a tag based on the `SHA256 digest`?

```shell
docker pull harbor.dockerregistry.com/kubernetes/pause@sha256:25e2e7fb27c4444cee9cccc0643befd31dbcc14821ce595943bc10304da28c53

docker pull harbor.dockerregistry.com/kubernetes/pause:3.9
```

![image](https://github.com/goharbor/harbor/assets/71524776/f15df59a-f5cf-4626-9910-60e3f75f20f1)

**Comment by diniramdhni** [comment]
Such functionalities are frequently implemented for user convenience by DockerHub and other container registries. If you use a self-hosted container registry or a registry that does not support this capability, you may need to manually create the pull command using the tag or digest. If you want to add equivalent capabilities to your own container registry's web interface, you must edit the registry's UI code. This would most likely entail frontend development work and would vary based on your container registry's technology stack and implementation specifics. If you use an open-source registry solution, you may consider contributing to its development or submitting a feature request for such an upgrade.

**Comment by ghost** [comment]
Got it, thank you for your answer.

**Comment by tostt** [comment]
In the example you highlight, there is only one tag, but generally you can have multiple tags for a single artifact.
The clickable icon can't know which tag you are interested in, so it provides a pull command for the artifact.
If you want to have distinct icons (one per tag) you need to click on the artifact sha256 item.
Then, you'll be taken to a page where you can click to retrieve a distinct pull command for each tag.

**Comment by AllForNothing** [comment]
@freemankevin Harbor already has supported such functionality:
if you want to `pull by sha256`,  use the copy button in the artifact list table:
![Screenshot 2023-12-11 at 16 54 16](https://github.com/goharbor/harbor/assets/30999793/0a08ba68-0fdd-4f1f-935a-227d29cb9667)
if you want to `pull by tag`, use the copy button in the artifact tag list table:
![Screenshot 2023-12-11 at 16 58 51](https://github.com/goharbor/harbor/assets/30999793/1edd803c-f496-40d8-8f96-03d6d1461c57)


note: the screenshots are from version 2.9.0 but you can still find similar ones in your version

**Comment by ghost** [comment]
Thank you, I really found this, but you need to click to the bottom to get the tag.

![image](https://github.com/goharbor/harbor/assets/71524776/c7c8a5e4-bcd4-4b9d-b964-1b4a468ae0be)

note: the screenshot is from version 2.8.4

**Comment by jameswsullivan** [comment]
In `v2.11.0` it's similar but need an extra step by selecting the tag and clicking the `COPY PULL COMMAND` dropdown (after clicking the link in the `Artifacts` column and land on the `Tags` page.

![harbor-pull-command](https://github.com/goharbor/harbor/assets/81947235/b31e1a92-5bd6-4deb-bf33-3293e33a7fec)
